src/autometrics/decorator.py

```diff
@@ -29,49 +36,103 @@ def autometrics(
     *,
     objective: Optional[Objective] = None,
 ):
+    def track_result_ok(start_time: float, function: str, module: str, caller: str):
+        get_tracker().finish(
+            start_time,
+            function=function,
+            module=module,
+            caller=caller,
+            objective=objective,
+            result=Result.OK,
+        )
+
+    def track_result_error(
+        exception: Exception,
+        start_time: float,
+        function: str,
+        module: str,
+        caller: str,
+    ):
+        get_tracker().finish(
+            start_time,
+            function=function,
+            module=module,
+            caller=caller,
+            objective=objective,
+            result=Result.ERROR,
+        )
+        # Reraise exception
+        raise exception
+
     """Decorator for tracking function calls and duration."""
```

Comment:
docstrings should be on the first line of the function. This docstring originally was meant to document what the `autometrics` function does (i.e. it is to be used as a decorator for your functions).

I understand it might be confusing because there's a function named `decorator` below. Maybe that should be renamed?

Reply:
yeah i'll rename it to sync_decorator just to be more explicit

Reply:
also added its own doc string